packages/starlight/components/Search.astro

```diff
@@ -86,6 +87,10 @@ const pagefindTranslations = {
 			openBtn.disabled = false;
 			closeBtn.addEventListener('click', closeModal);
 
+			dialog.addEventListener('close', () => {
+				document.body.toggleAttribute('data-search-modal-open', false);
```

Comment:
What do you think about moving all the teardown code in here?

```suggestion
				document.body.toggleAttribute('data-search-modal-open', false);
				window.removeEventListener('click', onClick);
```

And then `closeModal` gets simplified down to:

```js
			const closeModal = () => dialog.close();
```

Feels right to me to keep that code together to help maintain it all running together in the future.

Reply:
Makes sense, did both in one commit.